### PR TITLE
FIX: Blacklist nipype 1.2.1, 1.2.2 in Python 2.7 to avoid TraitError

### LINKED ISSUE #374: tests on python 2.7 started to fail within the past month
not sure yet what triggered it. https://travis-ci.org/nipy/heudiconv/builds/574912662 is the first build of master to exhibit it, but changes are unrelated.

```
=================================== FAILURES ===================================
_ test_reproin_largely_smoke[--files /home/travis/build/nipy/heudiconv/heudiconv/tests/data-reproin] _
/home/travis/build/nipy/heudiconv/heudiconv/tests/test_heuristics.py:72: in test_reproin_largely_smoke
    runner(args)
/home/travis/build/nipy/heudiconv/heudiconv/cli/run.py:128: in main
    process_args(args)
/home/travis/build/nipy/heudiconv/heudiconv/cli/run.py:335: in process_args
    dcmconfig=args.dcmconfig,)
/home/travis/build/nipy/heudiconv/heudiconv/convert.py:207: in prep_conversion
    dcmconfig=dcmconfig,)
/home/travis/build/nipy/heudiconv/heudiconv/convert.py:290: in convert
    bids_options, tmpdir, dcmconfig)
/home/travis/build/nipy/heudiconv/heudiconv/convert.py:439: in nipype_convert
    eg = convertnode.run()
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py:473: in run
    result = self._run_interface(execute=True)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py:564: in _run_interface
    return self._run_command(execute)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/pipeline/engine/nodes.py:671: in _run_command
    rebase=str2bool(self.config['execution']['use_relative_paths']))
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/pipeline/engine/utils.py:269: in save_resultfile
    setattr(result.outputs, key, val)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/nipype/interfaces/base/traits_extension.py:400: in validate
    value = super(MultiObject, self).validate(objekt, name, newvalue)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/traits/trait_types.py:2573: in validate
    return TraitListObject(self, object, name, value)
/home/travis/build/nipy/heudiconv/venv/lib/python2.7/site-packages/traits/trait_handlers.py:2474: in __init__
    raise excp
E   TraitError: Each element of the 'converted_files' trait of a Dcm2niixOutputSpec instance must be a pathlike object or string representing an existing file, but a value of '/tmp/dcm2niixRTkim9/convert/tmp/pytest-of-travis/pytest-0/test_reproin_largely_smoke___f0/Halchenko/Yarik/950_bids_test4/sub-phantom1sid1/ses-localizer/fmap/sub-phantom1sid1_ses-localizer_acq-3mm_phasediff_e2_ph.nii.gz' <class 'future.types.newstr.newstr'> was specified.
```

### PR DESCRIPTION
**Pull request by effigies**
First checking whether we've already fixed this in nipype `master`.

Fixes #374 
Related nipy/nipype#3037

### CODE REVIEW

**Comment by effigies** [comment]
K. Have to patch something quickly over at nipype, but this should be resolved for the next release.

**Comment by mgxd** [comment]
thanks, we'll leave the pin until the next release is out. just making a note here that we'll have to exclude nipype 1.2.2 for py2

**Comment by effigies** [comment]
I think you need to exclude 1.2.1 as well.

File: heudiconv/info.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 REQUIRES = [
     'nibabel',
     'pydicom',
-    'nipype>=1.0.0',
+    'nipype @ git+https://github.com/effigies/nipype.git@0196beac33d0c8c992bcc07b83c0b368810be02d',
```
Comment (mgxd) [inline heudiconv/info.py]: I think we'll want to leave nipype > 1.0.0 but add the pin for 2.7 only
```suggestion
    'nipype @ git+https://github.com/effigies/nipype.git@0196beac33d0c8c992bcc07b83c0b368810be02d; python_version == "2.7"',
```

Comment (effigies) [reply heudiconv/info.py]: I think the correct way would be:

```suggestion
    'nipype >=1.0.0; python_version > "3.0"',
    'nipype >=1.0.0,!=1.2.1,!=1.2.2; python_version == "2.7"',
```

Comment (mgxd) [reply heudiconv/info.py]: yep, I like that better and it won't require another PR 😄

**Comment by yarikoptic** [comment]
Thank you guys!

**Pull request merged**